docs/source/utilities/raster.rst

```diff
@@ -0,0 +1,58 @@
+Raster
+======
+
+Background
+----------
+
+This module extends xarray and rioxarray to provide
+additional functionality for raster processing and
+post-processing. These are used throughout the
+datasources to process various raster data, but are
+also useful directly for the end user, and is thus
+available for direct use. The extension is based on the
+guidance for how to `extend xarray
+<http://xarray.pydata.org/en/stable/internals/extending-xarray.html>`_.
+
+However, since rioxarray already extends xarray, this
+modules extensions inherit from the RasterArray and
+RasterDataset extensions respectively. This ensures
+cleaner code in the module as ``rio`` methods are
+available immediately, but also means a couple of
+design decisions are followed.
+
+The ``xarray.DataArray`` and ``xarray.Dataset``
+extensions here inherit from rioxarray base classes.
+Thus, methods that are identical for both objects
+are defined in a mixin class ``AatRasterMixin`` which
+can be inherited by the two respective extensions.
```

Comment:
I think the code implementation details can be kept to the source code, in the docs I would just focus on the usage (although indeed the complicated design does have impact on the functionality, so keep whatever you think is needed)

Reply:
Good point. I read through and don't think it has any info the user actually wants, because really they just care about how to access it not that it's an extension of `rioxarray` for instance which they'd probably access anyway using `rio.method()`